Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -2220,30 +2225,41 @@ void RenderFlexibleBox::alignChildren(const Vector<LineContext>& lineContexts)
                 continue;
             
             ItemPosition position = alignmentForChild(flexItem.box);
+            LayoutUnit maxAscent = position == ItemPosition::LastBaseline ? lineContext.lastBaselineMaxAscent : lineContext.maxAscent;
             if (position == ItemPosition::Stretch)
                 applyStretchAlignmentToChild(flexItem.box, lineCrossAxisExtent);
             LayoutUnit availableSpace = availableAlignmentSpaceForChild(lineCrossAxisExtent, flexItem.box);
-            LayoutUnit offset = alignmentOffset(availableSpace, position, marginBoxAscentForChild(flexItem.box), maxAscent, style().flexWrap() == FlexWrap::Reverse);
+            LayoutUnit offset = alignmentOffset(availableSpace, position, marginBoxAscentForChild(flexItem.box), maxAscent, style().flexWrap() == FlexWrap::Reverse, childIsParticipatingInBaselineAlignment(flexItem.box));
             adjustAlignmentForChild(flexItem.box, offset);
-            if (position == ItemPosition::Baseline && style().flexWrap() == FlexWrap::Reverse)
+            if ((position == ItemPosition::Baseline && containerHasWrapReverse) || (childIsParticipatingInBaselineAlignment(flexItem.box) && position == ItemPosition::LastBaseline && !containerHasWrapReverse)) {
+                needToAdjustItemsTowardsCrossAxis = true;
                 minMarginAfterBaseline = std::min(minMarginAfterBaseline, availableAlignmentSpaceForChild(lineCrossAxisExtent, flexItem.box) - offset);
+            }
         }
 
         minMarginAfterBaselines.append(minMarginAfterBaseline);
     }
-    
-    if (style().flexWrap() != FlexWrap::Reverse)
+
+    if (!needToAdjustItemsTowardsCrossAxis)
         return;
-    
+
+    auto adjustWrapReverseItem = [this, containerHasWrapReverse](RenderBox& box) { 
+        return containerHasWrapReverse && alignmentForChild(box) == ItemPosition::Baseline && !hasAutoMarginsInCrossAxis(box); 
+    };
+    auto adjustLastBaselineItem = [this, containerHasWrapReverse](RenderBox& box) { 
+        return childIsParticipatingInBaselineAlignment(box) && alignmentForChild(box) == ItemPosition::LastBaseline && !containerHasWrapReverse; 
```

Comment:
Why does `adjustLastBaselineItem` check `mainAxisIsChildInlineAxis(child)` while `adjustWrapReverseItem` which is for the 1st baseline doesn't check that?

Reply:
This was an attempt to deal with orthogonal items. This was for:

https://wpt.fyi/results/css/css-flexbox/flexbox-align-self-baseline-horiz-006.xh%5B…%5Droduct=safari&aligned&view=interop&q=label:interop-2021-flexbox?label=experimental&label=master&aligned

The comment in the test seemed to imply that orthogonal items should be ignored, but I learned that this was based off of an old version of the spec. Instead we should be synthesizing the baselines, but that looks like it will also require an in depth discussion. For now I'll be removing that check (since it shouldn't be there anyway) and will try to synthesize the baselines correctly in another patch.